pkg/controller/machinepool/ibmcloudactuator.go

```diff
@@ -0,0 +1,138 @@
+package machinepool
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	machineapi "github.com/openshift/api/machine/v1beta1"
+	ibmcloudprovider "github.com/openshift/cluster-api-provider-ibmcloud/pkg/apis/ibmcloudprovider/v1beta1"
+	installibmcloud "github.com/openshift/installer/pkg/asset/machines/ibmcloud"
+	installertypes "github.com/openshift/installer/pkg/types"
+	installertypesibmcloud "github.com/openshift/installer/pkg/types/ibmcloud"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/pkg/ibmclient"
+)
+
+// IBMCloudActuator encapsulates the pieces necessary to be able to generate
+// a list of MachineSets to sync to the remote cluster
+type IBMCloudActuator struct {
+	logger    log.FieldLogger
+	ibmClient ibmclient.API
+}
+
+var _ Actuator = &IBMCloudActuator{}
+
+func addIBMCloudProviderToScheme(scheme *runtime.Scheme) error {
+	return machineapi.AddToScheme(scheme)
+}
+
+// NewIBMCloudActuator is the constructor for building an IBMCloudActuator
+func NewIBMCloudActuator(ibmCreds *corev1.Secret, scheme *runtime.Scheme, logger log.FieldLogger) (*IBMCloudActuator, error) {
+	ibmClient, err := ibmclient.NewClientFromSecret(ibmCreds)
+	if err != nil {
+		logger.WithError(err).Warn("failed to create IBM client with creds in clusterDeployment's secret")
```

Comment:
Good point, I think it's a warn because we only need to reach out if `Zones` was unset (it most likely will be unset). Can improve the logic around that and return an error.

Reply:
I'm going to leave this as a warn in this PR. I'd like to tackle the other actuators at the same time if we want to improve that.